### PR TITLE
Fix to unblock Event Log failure in runtime:4.8-windowsservercore-ltsc2019 alt

### PR DESCRIPTION
**Pull request by bobsira**
Error:
error_2019

New Fix:
fix_2019_alt

This is a quick fix for this bug : https://github.com/microsoft/windows-container-tools/issues/150

All the steps to repro this are in the link above (the image affected, Dockerfile and LogMonitorConfig.json.

Build without the fix:
[LogMonitor.zip](https://github.com/microsoft/windows-container-tools/files/12715852/LogMonitor.zip)

Build with the fix:
[LogMonitorWithFix.zip](https://github.com/microsoft/windows-container-tools/files/12715806/LogMonitorWithFix.zip)

### CODE REVIEW

File: LogMonitor/src/LogMonitor/EventMonitor.cpp [hunk 1]
```diff
@@ -599,18 +599,152 @@ EventMonitor::EnableEventLogChannels()
...
+        }
+
+        
+
     }
 }
 
 
+// Converts the time to wait to a large integer
+LARGE_INTEGER EventMonitor::_ConvertWaitIntervalToLargeInt(int timeInterval)
```
Comment (iankingori) [inline LogMonitor/src/LogMonitor/EventMonitor.cpp:716]: This should probably be in utilities

Comment (bobsira) [reply LogMonitor/src/LogMonitor/EventMonitor.cpp:716]: Polishing up on this!

**Comment by TinaMor** [comment]
LGTM

**Review by iankingori — APPROVED** [review]

File: LogMonitor/src/LogMonitor/EventMonitor.cpp [hunk 2]
```diff
@@ -597,20 +597,109 @@ EventMonitor::PrintEvent(
...
+            HANDLE timerEvent = CreateWaitableTimer(NULL, FALSE, NULL);
+
+
+            if (!timerEvent) {
+            status = GetLastError();
+            logWriter.TraceError(
+                Utility::FormatString(
+                    L"Failed to create timer object.", status).c_str());
+            break;
+                        }
```
Comment (profnandaa) [inline LogMonitor/src/LogMonitor/EventMonitor.cpp:613]: lint: the curly bracket indent

File: LogMonitor/src/LogMonitor/Utility.h [hunk 3]
```diff
@@ -58,9 +61,20 @@ class Utility final
         _In_ const std::wstring& To
     );
 
-    static bool isJsonNumber(_In_ std::wstring& str);
+    static bool isJsonNumber(
+        _In_ std::wstring& str);
+
+    static void SanitizeJson(
```
Comment (profnandaa) [inline LogMonitor/src/LogMonitor/Utility.h:67]: wondering how this change came by?

**Review by profnandaa — APPROVED** [review]
Rest looks good to me, just a few comments to check out.

File: LogMonitor/src/LogMonitor/Utility.h (hunk 3 above)
Comment (bobsira) [reply LogMonitor/src/LogMonitor/Utility.h:67]: I must have move the parameters to the next line to match the way other functions have been authored in the file.

File: LogMonitor/src/LogMonitor/EventMonitor.cpp [hunk 4]
```diff
@@ -599,18 +599,119 @@ EventMonitor::EnableEventLogChannels()
...
+
+            std::double_t waitInSeconds = 300;
+
+            int elapsedTime = 0;
+
+            const int eventsCount = 2;
+            HANDLE channelEnableEvents[eventsCount] = {m_stopEvent, timerEvent};
+
+            while (elapsedTime < waitInSeconds) {
+
```
[the remark this answers is not in the document]
Comment (bobsira) [reply LogMonitor/src/LogMonitor/EventMonitor.cpp]: Done!

File: LogMonitor/src/LogMonitor/EventMonitor.cpp (hunk 2 above)
Comment (bobsira) [reply LogMonitor/src/LogMonitor/EventMonitor.cpp:613]: Fixed this!

**Review by iankingori — APPROVED** [review]

**Review by profnandaa — APPROVED** [review]

**Pull request merged**